unison-core/src/Unison/Name.hs

```diff
@@ -46,16 +47,12 @@ data Name
 
 toText :: Name -> Text
 toText (Name name) = name
+toText (Name' names) =
```

Comment:
This is kinda sketchy but I think it preserves how "name to text" currently works. We just puts dots between the name segments. I guess a `NameSegment` must not contain a `.` character? (If so that should probably be written down somewhere :))

Reply:
Side note: Not sure where this gets used, but it seems syntax-specific (#499).  

IMO, `NameSegment` should be allowed to have a `.`, but we'd need a way to escape / unescape the `NameSegment` according to the syntax in use.

Reply:
Totally agree.

Reply:
Right now, a tangle of parsers enforces the "no `.` in `NameSegment`" rule.